### PR TITLE
BE: metrics client

### PR DESCRIPTION
**Pull request by smonero**
### Description

RELIES ON https://github.com/lyft/clutch/pull/2246 to be done first.

Implementation of the BE for the metrics client. The metrics client is a service that passes received request queries from the FE to a prometheus endpoint and returns the result. The prometheus api endpoint is read from the config.

I also change the protos slightly to be closer to the prometheus api.

### Testing Performed

tested locally in private with curl

### GitHub Issue

Fixes #

### TODOs

- [x] add the mocks

### CODE REVIEW

**Comment by katpioro** [comment]
are these [the docs ](https://github.com/lyft/clutch-private#mocking-api-requests) related to how you tested this locally in private? additionally, what does "BE" stand for?

**Comment by smonero** [comment]
> 
BE means back-end. FE means front-end (I guess some people spell it frontEnd or front end).

yes those docs are useful for testing locally in private (I tested using the dev server not the mock server). To test in private it relies on: changing the config to have both the new service and module, and adding the prometheus api endpoint. In this PR in private https://github.com/lyft/clutch-private/pull/1503 I did those things but didn't include the endpoint.

**Comment by dschaller** [comment]
Can the protos be broken in to a separate PR? Can you link to the promethius API docs for the changes referenced?

File: api/config/service/metrics/v1/metrics.proto [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+package clutch.config.service.metrics.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/config/service/metrics/v1;metricsv1";
+
+import "validate/validate.proto";
+
+message Config {
+  // This should be an endpoint that mirrors the Prometheus API
+  string prometheus_api_endpoint = 1 [ (validate.rules).string = {min_bytes : 1} ];
```
Comment (dschaller) [inline api/config/service/metrics/v1/metrics.proto:11]: Would these changes to the protos to be close to promethius tie us to promethius?

Comment (smonero) [reply api/config/service/metrics/v1/metrics.proto:11]: yes. Is that a bad thing?

**Comment by smonero** [comment]
> Can the protos be broken in to a separate PR? Can you link to the promethius API docs for the changes referenced?

yeah here you go https://github.com/lyft/clutch/pull/2246/files

**Pull request closed without merging**